back-end/application.js

```diff
@@ -0,0 +1,59 @@
+// load .env data into process.env
+// require("dotenv").config();
+
+// Web server config
+const express = require("express");
+const bodyParser = require("body-parser");
+const sass = require("node-sass-middleware");
+const cors = require("cors");
+const morgan = require("morgan");
+const session = require("express-session");
+const app = express();
+
+const registerRoutes = require("./src/routes/register");
+const loginRoutes = require("./src/routes/login");
+const editProfileRoutes = require("./src/routes/editProfile");
+const questRoutes = require("./src/routes/quests");
+const userRoutes = require("./src/routes/users");
+const badgeRoutes = require("./src/routes/badges");
+const classRoutes = require("./src/routes/classes");
+
+module.exports = function application(ENV) {
```

Comment:
I personally prefer named export as opposed to default export. I like when things are explicit. I find it less prone to errors.

For instance if the implementation of the module's default exported thing changes, then the client code (the code relying on that exported thing) might get a bad surprise. However if the client code uses an exported function called `getApp()`, then at least 2 things could occur:
* if that function changes its name (thereby changing its semantics), that might mean the implementation of that function might not be adequate for the client code anymore, and it will fail fast which is what you want – it will not fail silently, which is the worst.
* if there's a refactor of that `application.js` module which would change the default exported thing to something entirely different, you'd be immune to that, since you only care about the named export `getApp()`.

But that's a matter of personal preference I guess.